File: codechef/hmappy2.py
def gcd(a,b):
	if b==0:
		return a
	return gcd(b,a%b)

t = int(input())
while t>0:
	t = t - 1
	n,a,b,k = map(int,input().split())
	g = (a*b)/gcd(a,b)
	c = (n//a) + (n//b) - 2*(n//g)
	if c>=k:
		print("Win")
	else:
		print("Lose")

File: codechef/hmappy2.cpp
#include<bits/stdc++.h>

using namespace std;

long long int gcd(long long int a,long long int b){
	return b==0?a:gcd(b,a%b);
}

int main(){
	int T;
	cin >> T;
	while(T--){
		long long int n,a,b,k;
		cin >> n >> a >> b >> k;
		long long int c1,c2;
		c1 = n/a - (n*gcd(a,b)/(a*b));
		c2 = n/b - (n*gcd(a,b)/(a*b));
		// cout << c1 << " " << c2 << endl;
		if((c1 + c2)>=k)
			cout << "Win" << endl;
		else
			cout << "Lose" << endl;
	}
	return 0;
}